### PR TITLE
OPL-11128 - Update Delete Documentation

### PR DESCRIPTION
**Pull request by pamaddox**
Much of the data delete documentation was either out of date or incorrect.

Removed the NNT delete documentation as that is no longer used and updated the full delete documentation to be correct.

It is likely there is still some incorrect data in the text and histogram delete calls, as things have changed a lot without changing the existing documentation, but this fixes the known issues that I found.

### CODE REVIEW

File: irondb/api/data-deletion.md [hunk 1]
```diff
@@ -655,69 +471,72 @@ DELETE
...
  * `query` : See [Tag Queries](/irondb/metric-names-tags-queries/#tag-queries) for more info on tag queries.
 
 #### Headers
 
-Used only with wildcards or tag query:
+Used with all deletes::
  * `x-snowth-account-id: <account_id>` (required)
    * `account_id` The account to be searched using the wildcard pattern
- * `x-snowth-advisory-limit: <integer>|none` (optional, defaults to no limit if not present)
+Used on with tag query delete:
```
Comment (esproul) [inline irondb/api/data-deletion.md:482]:
```suggestion
Used with tag query delete:
```

File: irondb/api/data-deletion.md [hunk 2]
```diff
@@ -655,69 +471,72 @@ DELETE
...
+ * `x-snowth-advisory-limit: <integer>|none` (optional, defaults to 100,000 if not present)
    * `integer` A positive integer specifying the number of matching results to
-     delete. If the header is unset, or set to -1 or "none", the service will
-     not limit the result set.
+     delete. If the header is unset, the default of 100,000 will be used.
+Used only with canonical metric delete:
```
Comment (esproul) [inline irondb/api/data-deletion.md:486]: This isn't used with all types of deletes?

File: irondb/api/data-deletion.md [hunk 3]
```diff
@@ -655,69 +471,72 @@ DELETE
...
 
-### Wildcard Metric Example
+### Query Example
 
 ```
 curl -X DELETE \
      -H 'x-snowth-account-id: 1234' \
      -H 'x-snowth-confirm-delete: 1' \
-     http://127.0.0.1:8112/full/6f6bdc73-2352-4bdc-ab0e-72f66d0dee12/multiple_example*
+     http://127.0.0.1:8112/full/tags/query=and(__check_uuid:6f6bdc73-2352-4bdc-ab0e-72f66d0dee12,__name:multiple_example*)
```
Comment (esproul) [inline irondb/api/data-deletion.md:517]:
```suggestion
     http://127.0.0.1:8112/full/tags?query=and(__check_uuid:6f6bdc73-2352-4bdc-ab0e-72f66d0dee12,__name:multiple_example*)
```

**Review by esproul — CHANGES_REQUESTED** [review]

File: irondb/api/data-deletion.md (hunk 2 above)
Comment (pamaddox) [reply irondb/api/data-deletion.md:486]: `full/canonical` is specifically for deleting a single metric. It matches a UUID and canonical metric name, which means that it is by definition only one metric.

Comment (esproul) [reply irondb/api/data-deletion.md]: I misunderstood, not enough surrounding lines. I thought you were saying that the confirm-delete header was only for canonical, but it's whatever comes _after_ that "only with canonical" line. My mistake.

Comment (pamaddox) [reply irondb/api/data-deletion.md]: Have changed to this:

```
Used with all deletes:
 * `x-snowth-account-id: <account_id>` (required)
   * `account_id` The account to be searched using the wildcard pattern
Used only with tag query delete:
 * `x-snowth-advisory-limit: <integer>|none` (optional, defaults to 100,000 if not present)
   * `integer` A positive integer specifying the number of matching results to
     delete. If the header is unset, the default of 100,000 will be used.
 * `x-snowth-confirm-delete: <0 or 1>` (optional, must be present and set to 1 to actually confirm and process the deletion)
```
I believe this to be correct.

**Review by esproul — APPROVED** [review]
LGTM, thanks for updating this.

**Pull request merged**